File: EscapeFromTrinityEngineStats/Models/RoomRecord.cs
using StatisticsApi.Models.Instances;

namespace StatisticsApi.Models
{
    public class RoomRecord
    {
        public int Id { get; set; }
        public virtual BattleRecord? BattleInstance { get; set; }
        public virtual EventRecord? EventInstance { get; set; }
        public virtual ShopRecord? ShopRecord { get; set; }
        public virtual RewardRecord? RewardRecord { get; set; }
    }
}

File: EFTEDashboard/src/app/interfaces/room-record.ts
import { BattleRecord } from "./battle-record";
import { EventRecord } from "./event-record";
import { RewardRecord } from "./reward-record";
import { ShopRecord } from "./shop-record";

export interface RoomRecord {
  id?: number;
  floorNumber?: number;
  levelNumber?: number;
  battleRecord?: BattleRecord | null;
  eventRecord?: EventRecord | null;
  shopRecord?: ShopRecord | null;
  rewardRecord?: RewardRecord | null;
}

